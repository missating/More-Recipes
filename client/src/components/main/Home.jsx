import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

// components
import { ConnectedSignup } from '../auth/Signup';
import { ConnectedSignin } from '../auth/Signin';
import { ConnectedTopRecipes } from '../recipe/TopRecipes';

export const Home = props => (
  <div>
    <div>
      <div className="bg flex">
        <div className="home-content center">
          <h1> More Recipes </h1>
          <p>A platform for you to share your awesome recipes </p>
          <div className="btn-group">
            {!props.authenticated && (
              <div>
                <a
                  href="#signup-modal"
                  data-toggle="modal"
                  data-target="#signup-modal"
                >
                  <button
                    type="button"
                    className="btn signup-button"
                  >
                    SIGN UP
                  </button>
                </a>
              </div>
            )}
            <ConnectedSignup />

            {!props.authenticated && (
              <div>
                <a
                  href="#signin-modal"
                  data-toggle="modal"
                  data-target="#signin-modal"
                >
                  <button
                    type="button"
                    className="btn signin-button"
                  >
                    SIGN IN
                  </button>
                </a>
              </div>
            )}
            <ConnectedSignin />
          </div>
        </div>
      </div>
      <div className="scroll-holder">
        <a
          href="#featured"
        >
          <i
            className="fas fa-angle-double-down fa-3x scroll-down"
          />
        </a>
      </div>
    </div>
    <ConnectedTopRecipes />
  </div>
);

Home.propTypes = {
  authenticated: PropTypes.bool.isRequired
};

const mapStateToProps = state => ({
  authenticated: state.auth.isAuthenticated,
});

const ConnectedComponent =
  connect(mapStateToProps)(Home);
export { ConnectedComponent as ConnectedHome };

