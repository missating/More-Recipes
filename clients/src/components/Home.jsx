import React from 'react';
import RecipeCard from './RecipeCard';
import { connect } from 'react-redux';
import Signup from './Signup';
import banner1 from '../assets/banner1.jpg';
import eight from '../assets/eight.jpg';
import seventen from '../assets/seventen.jpg';

import '../css/style.css';

class Home extends React.Component {
    render() {
        return (
    <div>
            <section className="container" style={{ marginTop: '50px' }}>
                <div className="row">
                    <div className="col-md-8">
                        <img src={banner1} alt="..." className="img-thumbnail size" id="mainImage"/>
                    </div>
                    <div className="col-md-4" id="media">
                        <p>More-Recipes provides a platform for users to share the awesome and exciting recipe ideas they have invented or learnt.</p>
                        <p>Suppose a user comes up with a recipe, he/she can post it on More-Recipes and get feedback in form of reviews and votes from other users who explore that recipe. Users can also keep a list of their favorite recipes on the application.</p>
                        <p><strong>Don't have an account ? </strong><a href="#myModal" data-toggle="modal" data-target="#myModal"><button type="button" className="btn btn-dark btn-sm fa fa-sign-in"> Sign Up</button></a></p>
                        <Signup />
                    </div>
                </div>
            </section>
        
        <section className="container" style={{ marginTop: '56px' }}>
          <h2 className="text-center"> Featured Recipes </h2>
          <hr/>
          <div className="row">
              <div className="col-md-4">
              <RecipeCard />
              </div>
              <div className="col-md-4">
              <RecipeCard />
              </div>
              <div className="col-md-4">
              <RecipeCard />
              </div>
            </div>
        
            <div className="row">
            <div className="col-md-4">
              <RecipeCard />
              </div>
              <div className="col-md-4">
              <RecipeCard />
              </div>
              <div className="col-md-4">
              <RecipeCard />
              </div> 
            </div>
            </section>
    </div>
        )
    }
}

export default connect(null)(Home);
