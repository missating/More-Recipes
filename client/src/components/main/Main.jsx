import React from 'react';
import { Switch, Route } from 'react-router-dom';
import { ConnectedHome } from './Home';
import { ConnectedRecipes } from '../recipe/Recipes';
import { ConnectedEditRecipe } from '../recipe/EditRecipe';
import { ConnectedSingleRecipe } from '../recipe/SingleRecipe';
import { ConnectedProfile } from '../user/Profile';
import { ConnectedAddRecipe } from '../recipe/AddRecipe';
import { ConnectedUserRecipes } from '../user/UserRecipes';
import { ConnectedUserFavourites } from '../user/UserFavourites';
import NotFound from '../common/NotFound';


export const Main = () => (
  <div className="main-cont">
    <Switch>
      <Route exact path="/" component={ConnectedHome} />
      <Route path="/recipes" exact component={ConnectedRecipes} />
      <Route path="/recipes/view/:id" exact component={ConnectedSingleRecipe} />
      <Route path="/profile" exact component={ConnectedProfile} />
      <Route path="/recipe/add" exact component={ConnectedAddRecipe} />
      <Route path="/users/recipes" exact component={ConnectedUserRecipes} />
      <Route path="/recipe/edit/:id" exact component={ConnectedEditRecipe} />
      <Route
        path="/users/favourites"
        exact
        component={ConnectedUserFavourites}
      />
      <Route component={NotFound} />
    </Switch>
  </div>
);

export { Main as RouterMain };
