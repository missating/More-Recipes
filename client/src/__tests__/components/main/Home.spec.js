import React from 'react';
import toJSON from 'enzyme-to-json';
import { Home } from '../../../components/main/Home';

describe('Home component', () => {
  const props = {
    authenticated: false
  };

  it('Should render', () => {
    const wrapper = shallow(<Home
      {...props}
    />);
    expect(wrapper.find('div').length).toEqual(8);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});

