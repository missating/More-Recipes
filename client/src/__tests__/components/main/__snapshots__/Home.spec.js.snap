// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component Should render 1`] = `
<div>
  <div>
    <div
      className="bg flex"
    >
      <div
        className="home-content center"
      >
        <h1>
           More Recipes 
        </h1>
        <p>
          A platform for you to share your awesome recipes 
        </p>
        <div
          className="btn-group"
        >
          <div>
            <a
              data-target="#signup-modal"
              data-toggle="modal"
              href="#signup-modal"
            >
              <button
                className="btn signup-button"
                type="button"
              >
                SIGN UP
              </button>
            </a>
          </div>
          <Connect(Signup) />
          <div>
            <a
              data-target="#signin-modal"
              data-toggle="modal"
              href="#signin-modal"
            >
              <button
                className="btn signin-button"
                type="button"
              >
                SIGN IN
              </button>
            </a>
          </div>
          <Connect(Signin) />
        </div>
      </div>
    </div>
    <div
      className="scroll-holder"
    >
      <a
        href="#featured"
      >
        <i
          className="fas fa-angle-double-down fa-3x scroll-down"
        />
      </a>
    </div>
  </div>
  <Connect(TopRecipes) />
</div>
`;
